### PR TITLE
Fixed build

### LINKED ISSUE #684: Documentation is broken
Build is not passing, because something is wrong with sphinx

see sphinx-doc/sphinx#2264

### PR DESCRIPTION
**Pull request by greg0ire**
Fixes #684

### CODE REVIEW

**Comment by OskarStark** [comment]
can you add line breaks? IMO its more readable

**Comment by greg0ire** [comment]
Sure!

**Comment by OskarStark** [comment]
@greg0ire can you please have a look why travis fails?

**Comment by greg0ire** [comment]
There are several problems, and I won't be able to work on them very soon.

One of them is about ACLs again, not sure if they are really required, I'll try to remove them.

**Comment by greg0ire** [comment]
I don't think there is a stable release of FOSUserBundle compatible with 3.0. I'll move the 3.0 builds for to the allowed_failures.

**Comment by greg0ire** [comment]
The 5.6 build fails with this message : 

> PHP Fatal error:  Call to undefined method Mock_Form_86db2601::bind() in /home/travis/build/sonata-project/SonataUserBundle/Controller/Api/GroupController.php on line 196

**Comment by greg0ire** [comment]
The last issue, with the security context class and interface being removed is harder than expected, I won't be able to fix it soon. Should I remove this particular build from the build matrix? I'm afraid if I allow 5.6 to fail, all builds with 5.6 will be allowed to fail.

File: .travis.yml [hunk 1]
```diff
@@ -34,16 +34,17 @@ matrix:
     - php: 5.6
       env: SYMFONY_VERSION=2.8.*
     - php: 5.5
-      env: SYMFONY_VERSION="3.0.*" ACL_VERSION="dev-master"
+      env: SYMFONY_VERSION="3.0.*"
```
Comment (core23) [inline .travis.yml]: You can remove the quotes too

**Comment by OskarStark** [comment]
allow failures for 5.6 is not an option IMO

**Comment by greg0ire** [comment]
@OskarStark : can you have a look at the 5.6 build though ? I'm under the impression that some part of symfony packages is in 3.0, and the other part in < 3.0

**Comment by greg0ire** [comment]
@OskarStark : the build won't run for 5.6 unless we restrict symfony to 2.8, which corresponds to another build that already exists. In other words, 5.6 will not pass until 5.6 + sf 3.0 passes :/ What should I do?

File: Tests/Form/Type/SecurityRolesTypeTest.php [hunk 2]
```diff
@@ -68,12 +69,26 @@ public function testGetName()
     public function testGetParent()
     {
         $type = new SecurityRolesType($this->roleBuilder);
-        $this->assertEquals('choice', $type->getParent());
+        $this->assertEquals(
+            version_compare(Kernel::VERSION, '2.8.0', '>=') ?
```
Comment (Soullivaneuh) [inline Tests/Form/Type/SecurityRolesTypeTest.php]: Don't use kernel version compare for this.

File: Tests/Form/Type/SecurityRolesTypeTest.php [hunk 3]
```diff
@@ -68,12 +69,26 @@ public function testGetName()
...
+                'Symfony\Component\Form\Extension\Core\Type\ChoiceType' :
+                'choice',
+            $type->getParent()
+        );
+    }
+
+    private function getSecurityRolesTypeName()
+    {
+        if (version_compare(Kernel::VERSION, '2.8.0', '>=')) {
```
Comment (Soullivaneuh) [inline Tests/Form/Type/SecurityRolesTypeTest.php]: Same as https://github.com/sonata-project/SonataUserBundle/pull/690/files#r52998441

File: Tests/Form/Type/SecurityRolesTypeTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Form/Type/SecurityRolesTypeTest.php]: Right, I'll use the same trick as in the other P.R. (with `setDefaultOptions`)

File: Form/Type/SecurityRolesType.php [hunk 4]
```diff
@@ -116,7 +117,11 @@ public function configureOptions(OptionsResolver $resolver)
      */
     public function getParent()
     {
-        return 'choice';
+        if (version_compare(Kernel::VERSION, '2.8.0', '>=')) {
```
Comment (Soullivaneuh) [inline Form/Type/SecurityRolesType.php]: https://github.com/sonata-project/SonataUserBundle/pull/690/files#r52998441

File: .travis.yml [hunk 5]
```diff
@@ -34,16 +34,17 @@ matrix:
...
     - php: 5.6
-      env: SYMFONY_VERSION="3.0.*" ACL_VERSION="dev-master"
+      env: SYMFONY_VERSION=3.0.*
     - php: 7.0
-      env: SYMFONY_VERSION="3.0.*" ACL_VERSION="dev-master"
+      env: SYMFONY_VERSION=3.0.*
 
   allow_failures:
     - php: hhvm
     - php: 5.3
```
Comment (Soullivaneuh) [inline .travis.yml]: Can someone explain me why the thing PHP 5.3 and 5.4 are on `allowed_failures`? O_O

BTW, test passes on those versions.

**Comment by Soullivaneuh** [comment]
I think we should add `conflicts` section for Symfony 3 and leave travis clear (PHP versions without SF version, and only one SF 3 build on `allowed_failures`).

Then, work on another PR for SF 3 compatibility.

Maybe a conflict like `"symfony/symfony": ">=3.0"` will works but not sure.

File: .travis.yml (hunk 5 above)
Comment (greg0ire) [inline .travis.yml]: Comes from b45ef9ce76ce22ab6c18c4ad927a8d0c9f261c4c by @core23 …

**Comment by greg0ire** [comment]
Let's give it a try

**Comment by greg0ire** [comment]
`conflict` does not seem to work as expected, probably because we are not really requiring `symfony/symfony` here

**Comment by greg0ire** [comment]
@Soullivaneuh : wouldn't it be better to simply remove the "|^3.0" for now?

**Comment by Soullivaneuh** [comment]
> `conflict` does not seem to work as expected, probably because we are not really requiring `symfony/symfony` here

That's too bad that composer does not handle reverse resolution for conflicts.

> @Soullivaneuh : wouldn't it be better to simply remove the `|^3.0` version constraint parts for now?

Maybe, but you'll have to change each dependencies on Travis to test 3.0.

**Comment by greg0ire** [comment]
> Maybe, but you'll have to change each dependencies on Travis to test 3.0.

Right, I'll do that.

**Comment by greg0ire** [comment]
It looks better. I'll try to replace the `version_compare` calls soon.

File: .travis.yml [hunk 6]
```diff
@@ -53,7 +50,14 @@ before_script:
   - composer global require phpunit/phpunit:@stable fabpot/php-cs-fixer --no-update
   - composer global update --prefer-dist --no-interaction
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update; fi;
-  - if [ "$ACL_VERSION" != "" ]; then composer require "symfony/security-acl:${ACL_VERSION}" --no-update; fi;
+  - |
+      if [ "$SYMFONY_VERSION" == "3.0.*" ]
```
Comment (Soullivaneuh) [inline .travis.yml]: `if; then` it more used AFAIK.

Comment (greg0ire) [inline .travis.yml]: Ok changing that.

**Comment by greg0ire** [comment]
@Soullivaneuh : replaced the `version_compare` calls, what do you think?

File: Form/Type/SecurityRolesType.php [hunk 7]
```diff
@@ -116,7 +116,10 @@ public function configureOptions(OptionsResolver $resolver)
      */
     public function getParent()
     {
-        return 'choice';
```
Comment (Soullivaneuh) [inline Form/Type/SecurityRolesType.php]: Please a comment with which SF version we are supporting here.

**Comment by Soullivaneuh** [comment]
Just one asking and we are good to merge IMO. :+1:

**Comment by greg0ire** [comment]
I'm on it!

**Comment by Soullivaneuh** [comment]
And OFC, squash your commits. ;-)

**Comment by greg0ire** [comment]
Do you know [my feelings about that](https://github.com/sonata-project/SonataPageBundle/pull/608#issuecomment-179477513)? I think I might be able to squash some of the commits, but not all of them.

**Comment by Soullivaneuh** [comment]
> Do you know my feelings about that? I think I might be able to squash some of the commits, but not all of them.

I globally agree for multiple commits if each commit is related to **only** one work.

For me, it should be one PR by work and this special case of PR "To make Travis green" should never happen.

This happened this time, but let's try to not have it again. ;-)

**Comment by greg0ire** [comment]
> this special case of PR "To make Travis green" should never happen.

There! That's why there are so many commits. Also, the PR contains some micro-work, that does not warrant a PR, but remains unrelated, like f6017f8 or 906a01b . It does not make the build green, but making a P.R. for that is well… a bit much.

**Comment by greg0ire** [comment]
I will try to squash 543eb17 and ab73141 though

**Comment by greg0ire** [comment]
Squashed the related commits, the tests are green, is this good to merge?

**Comment by OskarStark** [comment]
Thank you @greg0ire

**Pull request merged**